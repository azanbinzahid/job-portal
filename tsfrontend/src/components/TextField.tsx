import React, { FC } from "react";
import { Form } from "react-bootstrap";

type Props = {
  type: string;
  name: string;
  placeholder: string;
  value: string;
  onChange: any;
  error: any;
  label?: string;
};
const TextField: FC<Props> = ({
  type,
  name,
  placeholder,
  value,
  onChange,
  label,
  error,
}) => {
  return (
    <>
      <Form.Group>
        {label ? <Form.Label>{label}</Form.Label> : null}
        <Form.Control
          type={type}
          name={name}
          placeholder={placeholder}
          value={value}
          onChange={onChange}
        />
      </Form.Group>
      {error}
    </>
  );
};

export default TextField;
