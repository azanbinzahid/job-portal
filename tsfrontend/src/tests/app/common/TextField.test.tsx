import React from "react";
import { shallow } from "enzyme";
import TextField from "app/common/TextField";
test("renders the component", () => {
  const component = shallow(
    <TextField
      type="text"
      name="username"
      placeholder="Username"
      value="username"
      onChange={jest.fn()}
      error="error"
    />
  );
  expect(component).toMatchSnapshot();
});
