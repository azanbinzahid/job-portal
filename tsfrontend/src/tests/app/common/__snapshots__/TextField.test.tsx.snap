// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<Fragment>
  <FormGroup>
    <FormControl
      name="username"
      onChange={[MockFunction]}
      placeholder="Username"
      type="text"
      value="username"
    />
  </FormGroup>
  error
</Fragment>
`;
